app/components/navigation_header/index.tsx

```diff
@@ -60,46 +60,50 @@ const NavigationHeader = forwardRef<SearchRef, Props>(({
     const theme = useTheme();
     const styles = getStyleSheet(theme);
 
+    const [heightOffset, setHeightOffset] = React.useState(0);
```

Comment:
Since `heightOffset` in Header component is a number, and issue with shared value not updating on JS thread, I opt to use heightOffset as a component state instead.